### ISSUE TITLE
The plugin doesn't include my SVG symbols anymore

### ISSUE THREAD

**Issue by nhoizey** [brob/eleventy-plugin-svg-contents#7]
I use the plugin to build an SVG sprite from a set of symbols, and it doesn't work anymore with latest version.

[This Nunjucks template](https://github.com/nhoizey/nicolas-hoizey.com/blob/master/src/_includes/components/svg-sprite.njk) includes several symbols from SVG files:

```nunjucks
<svg style="display: none;" aria-hidden="true">
  {{ '/src/_assets/images/symbol-date.svg' | svgContents | safe }}
  {{ '/src/_assets/images/symbol-tags.svg' | svgContents | safe }}
  {{ '/src/_assets/images/symbol-webmentions.svg' | svgContents | safe }}
  {{ '/src/_assets/images/symbol-link.svg' | svgContents | safe }}
  {{ '/src/_assets/images/symbol-twitter.svg' | svgContents | safe }}
  {{ '/src/_assets/images/symbol-online.svg' | svgContents | safe }}
  {{ '/src/_assets/images/symbol-offline.svg' | svgContents | safe }}
  {{ '/src/_assets/images/symbol-info.svg' | svgContents | safe }}
</svg>
```

Here is [one of the symbols](https://raw.githubusercontent.com/nhoizey/nicolas-hoizey.com/master/src/_assets/images/symbol-date.svg):

```xml
<symbol id="symbol-date" viewBox="0 0 32 32" width="1em" height="1em">
<title>Date</title>
<path d="M29.334 3H25V1c0-.553-.447-1-1-1s-1 .447-1 1v2h-6V1c0-.553-.448-1-1-1s-1 .447-1 1v2H9V1c0-.553-.448-1-1-1S7 .447 7 1v2H2.667C1.194 3 0 4.193 0 5.666v23.667C0 30.806 1.194 32 2.667 32h26.667C30.807 32 32 30.806 32 29.333V5.666C32 4.193 30.807 3 29.334 3zM30 29.333c0 .368-.3.667-.666.667H2.667C2.3 30 2 29.7 2 29.333V5.666C2 5.3 2.3 5 2.667 5H7v2c0 .553.448 1 1 1s1-.447 1-1V5h6v2c0 .553.448 1 1 1s1-.447 1-1V5h6v2c0 .553.447 1 1 1s1-.447 1-1V5h4.334c.367 0 .666.3.666.666v23.667z"/>
<path d="M7 12h4v3H7v-3zm0 5h4v3H7v-3zm0 5h4v3H7v-3zm7 0h4v3h-4v-3zm0-5h4v3h-4v-3zm0-5h4v3h-4v-3zm7 10h4v3h-4v-3zm0-5h4v3h-4v-3zm0-5h4v3h-4v-3z"/>
</symbol>
```

Since [this commit](https://github.com/brob/eleventy-plugin-svg-contents/commit/eafe51d85b17eb5a783a598b7da1fd9a67d0aefd), the plugin uses Cheerio to load the SVG.

I guess Cheerio doesn't like my `symbol` that is not a proper SVG…

It works with 0.5.1 but not anymore with 0.6.0.

**Comment by brob** [comment]
Oh wow! This is a really interesting use case! The issue is in the first version, I was just grabbing the entire file's contents and in the new version I'm doing a look-up for the SVG tag, this was from a PR to have the ability to add a class to the svg. I'd never thought about someone using it in this way.

I'll get this fixed up.

**Comment by nhoizey** [comment]
I must admit it would be even better if I could have true SVG files, but include them as symbols (basically, replace `<svg…>` and `</svg>` with `<symbol…>` and `</symbol>`).

Could it be another option of your plugin, like the class addition?

**Comment by brob** [comment]
Here's a current proposed fix. Let me know what you think:

When wanting to return something other than an SVG element, you can provide a secondary argument of a selector to return.

```
    {{ "/sample/symbol.svg" | svgContents("classname", "symbol") | safe }}
    {{ "/sample/symbol.svg" | svgContents('', 'symbol') | safe }} // no class name
    {{ "/sample/symbol.svg" | svgContents('', '#idToGrab') | safe }}
```

**Comment by brob** [comment]
If no second argument is added, it'll default to grabbing the SVG element.

**Comment by nhoizey** [comment]
Looks perfect, I can add a `<svg>` around the `<symbol>` in my files! 👍

**Comment by brob** [comment]
It looks like it works, but if you've got the time to test a branch, I'd love to make sure it's meeting the use case's need before I publish.

it's on this branch: https://github.com/brob/eleventy-plugin-svg-contents/tree/optional-selector

If not, my testing shows it's working and I'm happy to bump a new version out there.

**Comment by brob** [comment]
Set up a beta version in NPM: https://www.npmjs.com/package/eleventy-plugin-svg-contents/v/0.7.2-beta.0

**Comment by brob** [comment]
> Looks perfect, I can add a `<svg>` around the `<symbol>` in my files! 👍

In your SVG files, you shouldn't need to wrap it in an SVG with this solution. You should just be able to pass the `'symbol'` string as the second option to the filter and it will just pull the symbol into your template.

**Comment by nhoizey** [comment]
It "works" with the beta, as it now includes my `<symbol>` images! 👍

However, if I write this:
```nunjucks
{{ '/src/_assets/images/symbol-date.svg' | svgContents('', 'symbol') | safe }}
```

And I change my `symbol-date.svg` file to add a `<svg>…</svg>` like this:

```svg
<svg xmlns="https://www.w3.org/2000/svg" viewBox="0 0 32 32">
  <symbol id="symbol-date" viewBox="0 0 32 32" width="1em" height="1em">
    <title>Date</title>
    <path d="M29.334 3H25V1c0-.553-.447-1-1-1s-1 .447-1 1v2h-6V1c0-.553-.448-1-1-1s-1 .447-1 1v2H9V1c0-.553-.448-1-1-1S7 .447 7 1v2H2.667C1.194 3 0 4.193 0 5.666v23.667C0 30.806 1.194 32 2.667 32h26.667C30.807 32 32 30.806 32 29.333V5.666C32 4.193 30.807 3 29.334 3zM30 29.333c0 .368-.3.667-.666.667H2.667C2.3 30 2 29.7 2 29.333V5.666C2 5.3 2.3 5 2.667 5H7v2c0 .553.448 1 1 1s1-.447 1-1V5h6v2c0 .553.448 1 1 1s1-.447 1-1V5h6v2c0 .553.447 1 1 1s1-.447 1-1V5h4.334c.367 0 .666.3.666.666v23.667z"/>
    <path d="M7 12h4v3H7v-3zm0 5h4v3H7v-3zm0 5h4v3H7v-3zm7 0h4v3h-4v-3zm0-5h4v3h-4v-3zm0-5h4v3h-4v-3zm7 10h4v3h-4v-3zm0-5h4v3h-4v-3zm0-5h4v3h-4v-3z"/>
  </symbol>
</svg>
```

The full SVG is inlined, not only the `<symbol>…</symbol>`.

Isn't the second argument supposed to "extract" the `<symbol>` from the `<svg>`?

**Comment by brob** [comment]
That's interesting... It's supposed to. It uses the second argument as a Cheerio Selector (which functions like an old-school jQuery selector... I'll poke it a bit more tonight. I hadn't tested with the SVG wrapped around the symbol. I'll do that.

**Comment by brob** [comment]
New beta version is live: https://www.npmjs.com/package/eleventy-plugin-svg-contents/v/0.7.2-beta.1

This SHOULD be working. I forgot about a conditional that your code was running into. That's cleared up and now tested against (trying to get into the habit of testing this stuff, since people are apparently actually using it :D)

Give that a shot and let me know

**Comment by nhoizey** [comment]
It works perfectly now! 👍

I'm also learning to make tests in my plugins… 😁

**Comment by nhoizey** [comment]
Sorry, I just noticed a new issue, but I don't really now if it comes from your plugin.

I use `<path d="M13 28.8…" />` elements in my SVG, and it looks like your plugin remove the auto-closing `/>` at the end.

To consecutive `<path /><path />` become parent/child (`<path><path>`) and rendering is broken.

**Comment by brob** [comment]
Hrmmmmm. So it looks like by default Cheerio doesn't like self-closing tags. There's supposed to be an option I can pass it that will recognize them, but it's not working as expected. I'll investigate.

**Comment by nhoizey** [comment]
Ok. This is really important for SVG, maybe Cheerio has a specific option for them.

**Comment by brob** [comment]
New beta version: https://www.npmjs.com/package/eleventy-plugin-svg-contents/v/0.7.2-beta.2

This should respect all self-closing tags. It's working on what you posted above.

This is what's showing up in my final source from beta 2:

```html
    <symbol id="symbol-date" viewBox="0 0 32 32" width="1em" height="1em">
    <title>Date</title>
    <path d="M29.334 3H25V1c0-.553-.447-1-1-1s-1 .447-1 1v2h-6V1c0-.553-.448-1-1-1s-1 .447-1 1v2H9V1c0-.553-.448-1-1-1S7 .447 7 1v2H2.667C1.194 3 0 4.193 0 5.666v23.667C0 30.806 1.194 32 2.667 32h26.667C30.807 32 32 30.806 32 29.333V5.666C32 4.193 30.807 3 29.334 3zM30 29.333c0 .368-.3.667-.666.667H2.667C2.3 30 2 29.7 2 29.333V5.666C2 5.3 2.3 5 2.667 5H7v2c0 .553.448 1 1 1s1-.447 1-1V5h6v2c0 .553.448 1 1 1s1-.447 1-1V5h6v2c0 .553.447 1 1 1s1-.447 1-1V5h4.334c.367 0 .666.3.666.666v23.667z"/>
    <path d="M7 12h4v3H7v-3zm0 5h4v3H7v-3zm0 5h4v3H7v-3zm7 0h4v3h-4v-3zm0-5h4v3h-4v-3zm0-5h4v3h-4v-3zm7 10h4v3h-4v-3zm0-5h4v3h-4v-3zm0-5h4v3h-4v-3z"/>
  </symbol>
```

**Comment by nhoizey** [comment]
It works indeed! 👍

**Comment by brob** [comment]
Feature merged in and published as 0.7.0 on NPM!

Thanks for all the testing!

**Comment by nhoizey** [comment]
You're welcome!
